src/redux/actions/planActions.js

```diff
@@ -0,0 +1,329 @@
+      method: 'POST',
+      body: JSON.stringify(plan)
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) dispatch(addPlanFailure());
+        else {
+          dispatch(addPlanSuccess());
+          dispatch(getPlanInfo(json.result)); // Fetch added plan
+        }
+      });
+  };
+}
+
+/**
+ * Performs an http PUT updatePlan request to server.
+ * Dispatches updatePlan to indicate the beginning of an updatePlan process.
+ * Dispatches updatePlanFailure to indicate the end of a failed updatePlan process.
+ * Dispatches updatePlanSuccess to indicate the end of a successful updatePlan process.
+ * If updatePlan process succeeds, the outdated plan is filtered out of the state list
+ * and getPlanInfo is dispatched using the e_id of the updated plan.
+ * @param plan object containing name, dscript, e_id.
+ * @returns {function(*): Promise<Response | never>} dispatch results.
+ */
+export function updatePlan(plan) {
+  return (dispatch, getState) => {
+    dispatch(updatePlanRequest()); // Update plan request process has begun...
+    return fetch('http://localhost:4201/plan/update', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'PUT',
+      body: JSON.stringify(plan)
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) dispatch(updatePlanFailure());
+        else {
+          // Delete outdated plan, fetch updated plan
+          let plans = getState().masterReducer.plans.plans;
+          delete plans[plan.trip_id][plan.e_id];
+          dispatch(updatePlanSuccess());
+          dispatch(getPlanInfo(plan.e_id));
+        }
+      });
+  };
+}
+
+/**
+ * Performs an http DELETE deletePlan request to server.
+ * Dispatches deletePlan to indicate the beginning of a deletePlan process.
+ * Dispatches deletePlanFailure to indicate the end of a failed deletePlan process.
+ * Dispatches deletePlanSuccess to indicate the end of a successful deletePlan process.
+ * If deletePlan process succeeds, the deleted plan is filtered out of the
+ * state plan list.
+ * @param plan object containing name, dscript, e_id.
+ * @returns {function(*): Promise<Response | never>} dispatch results.
+ */
+export function deletePlan(plan) {
+  return (dispatch, getState) => {
+    dispatch(deletePlanRequest()); // Delete plan request process has begun...
+    return fetch('http://localhost:4201/plan/delete', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'DELETE',
+      body: JSON.stringify(plan)
+    })
+      .then(response => response.json())
+      .then(json => {
+        if (json.success === false) dispatch(deletePlanFailure());
+        else {
+          // Deleted plan from state plan list
+          let plans = getState().masterReducer.plans.plans;
```

Comment:
getState again, leave it to the reducer (I think)

Reply:
See my comment above.